src/plugins/shiftmanager/shiftmanager.go

```diff
@@ -0,0 +1,296 @@
+/*
+ * Radon
+ *
+ * Copyright 2018-2020 The Radon Authors.
+ * Code is licensed under the GPLv3.
+ *
+ */
+
+package shiftmanager
+
+import (
+	"fmt"
+	"runtime"
+	"sync"
+
+	"github.com/radondb/shift/build"
+	"github.com/radondb/shift/shift"
+	sxlog "github.com/radondb/shift/xlog"
+	"github.com/xelabs/go-mysqlstack/sqlparser/depends/sqltypes"
+	"github.com/xelabs/go-mysqlstack/xlog"
+)
+
+const maxShiftInstancesAlived = 10
+
+// ShiftManager used to manager the infos about the shift
+// Called by reshard/rebalance
+type ShiftManager struct {
+	log *xlog.Log
+	mu  sync.Mutex
+	wg  sync.WaitGroup
+
+	// Limit max 10 shift instances
+	// key: for reshard, key is db_table, for rebalance, key is db_table_backend
```

Comment:
The key about db_table need fixed:
db：a_b, table:c
db：a, table:b_c
the key is the same?

Reply:
The key changed to string: `db`.`table`